nion/swift/ImageCanvasItem.py

```diff
@@ -349,8 +387,24 @@ def set_data_info(self, dimensional_calibration: Calibration.Calibration, info_i
             self.__update_sizing()
             self.update()
 
+    def set_display_properties(self, display_properties: Persistence.PersistentDictType) -> None:
+        scale_marker_position = display_properties.get("scale_marker_position", None)
+        background_fill_color = display_properties.get("scale_background_fill_color",None)
+
+        needs_update = (
+                scale_marker_position != getattr(self, "_scale_marker_position", None) or
+                background_fill_color != getattr(self, "_background_fill_color", None)
```

Comment:
I feel like I might have made this comment already - but can you make `scale_marker_position` and `background_fill_color` into properties and have the `setters` check whether they need update and call update? Then also use `self.background_fill_color` and `self.scale_marker_position` when constructing the `ScaleMarkerCanvasItemComposer` below? That eliminates use of un-typed `getattr` in both places.